osgbuild/svn.py

```diff
@@ -166,17 +166,30 @@ def restricted_branch_matches_target(branch, target):
     """
     for (branch_pattern, branch_name) in SVN_RESTRICTED_BRANCHES.items():
         branch_match = re.search(branch_pattern, branch)
+        if not branch_match:
+            continue
         for (target_pattern, target_name) in KOJI_RESTRICTED_TARGETS.items():
             target_match = re.search(target_pattern, target)
-
-            if branch_match and target_match and branch_name == target_name:
-                if branch_name == "versioned":
-                    return branch_match.group("osgver") == target_match.group("osgver")
-                elif branch_name == "upcoming":
-                    return ((branch_match.group("osgver") or "3.5") ==
-                            (target_match.group("osgver") or "3.5"))
-                else:
-                    return True
+            if not target_match:
+                continue
+
+            target_osgver = target_match.groupdict().get("osgver", None)
+            branch_osgver = branch_match.groupdict().get("osgver", None)
+            if branch_name == "main":
+                branch_osgver = MAIN_OSGVER
+            if target_name == "main":
+                target_osgver = MAIN_OSGVER
+            if branch_name == "upcoming":
+                branch_osgver = branch_osgver or UPCOMING_OSGVER
+            if target_name == "upcoming":
+                target_osgver = target_osgver or UPCOMING_OSGVER
+
+            if target_name in ["main", "versioned"] and branch_name in ["main", "versioned"]:
+                return branch_osgver == target_osgver
+            elif target_name == "upcoming" and branch_name == "upcoming":
+                return branch_osgver == target_osgver
```

Comment:
Are these `branch_osgver == target_osgver` expressions meant to handle the case where both values are `None`?

Reply:
Seems that this will return `True` even when comparing `main` and `versioned`, which does not seem (to me) to match the docstring for this function.  Am i reading it wrong, or does the docstring need to be updated?

Reply:
The docstring needs to be updated.

If `branch_name` is `"versioned"` then `branch_osgver` can't be `None` (or else it wouldn't match the pattern).
If `branch_name` is `"main"` then `branch_osgver` is assigned `MAIN_OSGVER` which is not `None`.

If `branch_name` is `"upcoming"` then `branch_osgver` is either not `None` or is assigned `UPCOMING_OSGVER` which is not `None`.

Same thing for `target_osgver`.

So none of the `branch_osgver == target_osgver` comparisons will be reached if either of those are `None`.